presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoResultSet.java

```diff
@@ -359,6 +359,15 @@ private Timestamp getTimestamp(int columnIndex, DateTimeZone localTimeZone)
             }
         }
 
+        if (columnInfo.getColumnTypeName().equalsIgnoreCase("timestamp with time zone")) {
+            ZonedDateTime zonedDateTime = getTimestampWithTimeZone(columnIndex);
+            verify(zonedDateTime != null);
+            // point in time
+            Timestamp timestamp = new Timestamp(zonedDateTime.toEpochSecond() * 1000);
```

Comment:
`SECONDS.toMillis(zonedDateTime.toEpochSecond())`

But how is this different than
```java
Timestamp.valueOf(zonedDateTime.toLocalDateTime())

Reply:
This is different semantics.

`Timestamp.valueOf(zonedDateTime.toLocalDateTime())` is like `java.time.ZonedDateTime#withZoneSameLocal`
current code is like `java.time.ZonedDateTime#withZoneSameInstant`